pkg/storage/storage_exemplars_query.go

```diff
@@ -0,0 +1,190 @@
+package storage
+
+import (
+	"context"
+	"time"
+
+	"github.com/pyroscope-io/pyroscope/pkg/flameql"
+	"github.com/pyroscope-io/pyroscope/pkg/storage/metadata"
+	"github.com/pyroscope-io/pyroscope/pkg/storage/tree"
+)
+
+type QueryExemplarsInput struct {
+	Query     *flameql.Query
+	StartTime time.Time
+	EndTime   time.Time
+
+	MinValue uint64
+	MaxValue uint64
+
+	HeatmapTimeBuckets  int64
+	HeatmapValueBuckets int64
+}
+
+type QueryExemplarsOutput struct {
+	Tree     *tree.Tree
+	Count    uint64
+	Heatmap  *Heatmap
+	Metadata metadata.Metadata
+
+	Telemetry map[string]interface{}
+}
+
+func (*Storage) QueryExemplars(_ context.Context, in QueryExemplarsInput) (QueryExemplarsOutput, error) {
+	// FIXME(kolesnikovae): Not implemented.
+	// Stub data for testing.
+
+	// The heatmap includes 10m of sample data (~100K exemplars).
+	// Note that the distribution of values is non-uniform:
+	// most of the values are concentrated in the bottom Y-bin,
+	// as exemplars typically include ver few samples (1-2).
+	h := &Heatmap{
+		TimeBuckets:  60,
+		ValueBuckets: 32,
+		StartTime:    time.Date(2022, time.August, 4, 21, 40, 0, 0, time.Local),
+		EndTime:      time.Date(2022, time.August, 4, 21, 50, 0, 0, time.Local),
+		MinValue:     1,
+		MaxValue:     110,
+		MinDepth:     1,
+		MaxDepth:     23077,
+		Values:       datasetA,
+	}
+
+	// If MaxValue was specified, we respond with another dataset:
+	// original min-max: (18:50]
+	if in.MaxValue > 0 {
+		h.MinValue = 19
+		h.MaxValue = 50
+		h.MinDepth = 1
+		h.MaxDepth = 21
+		h.Values = datasetB
+	}
+
+	return QueryExemplarsOutput{Tree: tree.New(), Heatmap: h}, nil
+}
```

Comment:
@dogfrogfog, please take a look – I added two data stubs for heatmap: 
- Both samples refer to a static 10 minutes range.
- By default, the dataset A is returned: this one is interesting in the way that its nature is super close to the real data. I imagine that colouring would be pretty tricky, I'd suggest considering logarithmic scale (worth discussing).
- In case if `maxValue` parameter is provided, the mock returns dataset B, that was crafted with MinValue 18 and MaxValue 50 (on the same data set as in the first case).

Below are reference visualisations (don't mind styling – I use this for debugging only):

Dataset A:
![image](https://user-images.githubusercontent.com/12090599/187900411-9cdd58b3-c1e7-4b00-a55d-cebb3606a272.png)

Dataset B:
![image](https://user-images.githubusercontent.com/12090599/187900352-644e2551-5bf0-420b-aebe-624d37ac41a7.png)


I hope you will find it helpful.

This stub will be removed before merge.

/cc @Rperry2174

Reply:
This request returns dataset A:
```
http://localhost:4040/api/exemplars:query?query=app{}
```
One for dataset B:
```
http://localhost:4040/api/exemplars:query?query=app{}&maxValue=0
```

Let me know if you need just a response JSON